docs/index.rst

```diff
@@ -19,16 +19,25 @@ In addition, it's possible to change the used backend via a context manager.
 Relation to the NumPy duck-array ecosystem
 ------------------------------------------
 
-:obj:`uarray` is a backend/dispatch mechanism with a focus on array computing and the
-needs of the wider array community, by allowing a clean way to register an
-implementation for any Python object (functions, classes, class methods, properties,
-dtypes, ...), it also provides an important building block for
-`NEP-22 <http://www.numpy.org/neps/nep-0022-ndarray-duck-typing-overview.html>`_.
-It is meant to address the shortcomings of `NEP-18
-<http://www.numpy.org/neps/nep-0018-array-function-protocol.html>`_ and `NEP-13
-<https://www.numpy.org/neps/nep-0013-ufunc-overrides.html>`_;
-while still holding nothing in :obj:`uarray` itself that's specific to array computing
-or the NumPy API.
+There are three main NumPy enhancement proposals (NEPs) inside NumPy itself that relate
+to the duck-array ecosystem. There is `NEP-22 <http://www.numpy.org/neps/nep-0022-ndarray-duck-typing-overview.html>`_,
+which is a high-level overview of the duck-array ecosystem, and the direction NumPy
+intends to move towards. Two main protocols were introduced to fill this gap,
+the ``__array_function__`` protocol defined in `NEP-18 <http://www.numpy.org/neps/nep-0018-array-function-protocol.html>`_,
+and the older ``__array_ufunc__`` protocol defined in `NEP-13 <https://www.numpy.org/neps/nep-0013-ufunc-overrides.html>`_.
+
+:obj:`unumpy` provides an an alternate framework based on :obj:`uarray`, bypassing
+the ``__array_function__`` and ``__array_ufunc__`` protocols entirely. It
+provides a clear separation of concerns. It defines callables which can be overridden,
+and expresses everything else in terms of these callables. See the :obj:`uarray` documentation
+for more details.
+
+For example, you can override ``__call__`` on a ``ufunc`` to override it. You can also override
+its methods, ``reduce``, and so on.
+
+Theoretically, you could convert a ``ufunc`` by overriding ``__get__``, but there's also a dedicated
```

Comment:
I'd remove this sentence, more confusing than helpful.

Reply:
Same here.